### PR TITLE
Set up basic build system. Add simple ```vkinfo``` app for displaying platform information

### PR DESCRIPTION
**Pull request by serjzimmerman**
- Set up build system for Vulkan with CMake
- Switch to clang-format
- Add CI check for code formatting on push and PRs into main
- Provide sample application that uses ```<vulkan.hpp>```

### CODE REVIEW

**Comment by serjzimmerman** [comment]
Maybe I should remove ```uncrustify.cfg``` as we've decided to stick with clang-format from now on?

**Comment by NikitaDzer** [comment]
> Maybe I should remove `uncrustify.cfg` as we've decided to stick with `clang-format` from now on?

Yep, remove it.

**Comment by krisszzzz** [comment]
Change ```clang-format``` settings that divide *member function* return type and the function name. According to the code style (see ```code-style``` branch)  requirements, we **should not divide** *member function* return type and the function name.

**Review by NikitaDzer — COMMENTED** [review]
Sergo, check it.

File: utils/vkinfo.cc [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+#include <fmt/core.h>
+
+#include <algorithm>
+#include <cstdint>
+#include <numeric>
+#include <sstream>
+#include <string>
+
+#include "unified_includes/vulkan_include.h"
```
[the remark this answers is not in the document]
Comment (serjzimmerman) [reply utils/vkinfo.cc]: Sure

**Comment by NikitaDzer** [comment]
Ship it!

**Review by krisszzzz — APPROVED** [review]
Ship it!

**Review by NikitaDzer — APPROVED** [review]

**Pull request merged**